src/PhpSpec/Formatter/TapFormatter.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Formatter;
+
+use PhpSpec\Event\SuiteEvent;
+use PhpSpec\Event\SpecificationEvent;
+use PhpSpec\Event\ExampleEvent;
+use Symfony\Component\Yaml\Yaml;
+
+class TapFormatter extends ConsoleFormatter
+{
+
+    const VERSION = 'TAP version 13';
+
+    const OK = 'ok %d';
+
+    const NOT_OK = 'not ok %d';
+
+    const DESC = ' - %s: %s';
+
+    const SKIP = ' # SKIP %s';
+
+    const TODO = ' # TODO %s';
+
+    const PLAN = '1..%d';
+
+    const MESSAGE = "---\nmessage: %s\n...";
+
+    const SEVERITY = "\n---\nseverity: %s\n...";
+
+    /**
+     * @var int
+     */
+    private $examplesCount = 0;
+
+    /**
+     * @var string
+     */
+    private $currentSpecificationTitle;
+
+    /**
+     * @param SuiteEvent $event
+     */
+    public function beforeSuite(SuiteEvent $event)
+    {
+        $this->getIO()->writeln(self::VERSION);
+    }
+
+    /**
+     * @param SpecificationEvent $event
+     */
+    public function beforeSpecification(SpecificationEvent $event)
+    {
+        $this->currentSpecificationTitle = $event->getSpecification()->getTitle();
+    }
+
+    /**
+     * @param ExampleEvent $event
+     */
+    public function afterExample(ExampleEvent $event)
+    {
+        $this->examplesCount++;
+        $desc = sprintf(
+            self::DESC,
+            $this->currentSpecificationTitle,
+            preg_replace('/^its? /', '', $event->getExample()->getTitle())
+        );
+
+        switch ($event->getResult()) {
+            case ExampleEvent::PASSED:
+                $result = sprintf(self::OK, $this->examplesCount) . $desc;
+                break;
+            case ExampleEvent::PENDING:
+                $message = $this->getResultData($event, $event->getResult());
+                $result = sprintf(self::NOT_OK, $this->examplesCount) . $desc . $message;
+                break;
+            case ExampleEvent::SKIPPED:
+                $message = sprintf(self::SKIP, $this->getResultData($event));
+                $result = sprintf(self::OK, $this->examplesCount) . $desc . $message;
+                break;
+            case ExampleEvent::BROKEN:
+            case ExampleEvent::FAILED:
+                $message = $this->getResultData($event, $event->getResult());
+                $result = sprintf(self::NOT_OK, $this->examplesCount) . $desc . "\n" . $message;
+                break;
+        }
+
+        $this->getIO()->writeln($result);
```

Comment:
I'm thinking more like future result types (Skipped was fairly recent for instance). In 2020 when we have quantum computing and someone adds 'ExampleEvent::UNCOLLAPSED' I don't want your formatter to break ;-)

 I think it should be 'not OK' so that someone comes and shouts at me.